test/end-to-end/test_pkg_download.sh

```diff
@@ -7,21 +7,26 @@
 # for cleaner code and more fine grained tests. However, they are a
 # bit of a pain to program in bash. This is intended to provide
 # minimal testing pending our figuring out the best approach for
-# command line testing. 
+# command line testing.
 #
+# Assumptions:
+# 1. HAB_AUTH_TOKEN Environment variables is set and valid
+# 2. ${CACHE_DIR} can be set to a writable location on the filesystem
 
 set -euo pipefail
 
 # Test the Habitat package downloader.
 #
 # Uses the `HAB_INTERNAL_BLDR_CHANNEL` environment variable to control
 # the base packages channel for the exporter.
-HAB=$HAB_TEST_CMD || "hab"
+HAB=${HAB_TEST_CMD:-hab}
 CACHE_DIR="test-cache"
 IDENT_FILE="ident_file"
+# we explicitly define --channel where needed in the tests
+unset HAB_BLDR_CHANNEL
```

Comment:
Also just to be clear, I think you only need this, and not also the `--channel=stable` options. Having both doesn't hurt, of course, but it's a bit of a "belt and suspenders" kind of thing :smile:

Reply:
Right, it does seem a little over zealous. I've removed the `unset` preferring the explicit `--channel stable` argument here https://github.com/habitat-sh/habitat/pull/7104/commits/f6d3c4bb03a461c5282e7607e61682d1cbedbb1d

I'll let the build go green then merge since it has two approvals. Thank you!